ex2_hal/athena/equipment_handler/source/ina209.c

```diff
@@ -0,0 +1,247 @@
+    return;
+}
+
+void ina209_set_configuration(uint8_t addr, uint16_t *val) {
+    ina209_Write2ByteReg(addr, 0x00, *val);
+    return;
+}
+
+void ina209_get_status_flags(uint8_t addr, uint16_t *retval) {
+    //  POR is x0000
+    ina209_Read2ByteReg(addr, 0x01, retval);
+    return;
+}
+
+// probably not needed; designs have alert pin grounded
+void ina209_get_control_register(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x02, retval);
+    return;
+}
+
+void ina209_set_control_register(uint8_t addr, uint16_t *val) {
+    ina209_Write2ByteReg(addr, 0x02, *val);
+    return;
+}
+
+void ina209_get_shunt_voltage(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x03, retval);
+    return;
+}
+
+void ina209_get_bus_voltage(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x04, retval);
+    return;
+}
+
+void ina209_get_power(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x05, retval);
+    return;
+}
+
+// Current defaults to 0 on POR before calibration register (x16) is ina209_set
+void ina209_get_current(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x06, retval);
+    return;
+}
+
+void ina209_get_shunt_voltage_peak_pos(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x07, retval);
+    return;
+}
+
+void ina209_get_shunt_voltage_peak_neg(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x08, retval);
+    return;
+}
+
+void ina209_get_bus_voltage_peak_max(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x09, retval);
+    return;
+}
+
+void ina209_get_bus_voltage_peak_min(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x0A, retval);
+    return;
+}
+
+void ina209_get_power_peak(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x0B, retval);
+    return;
+}
+
+void ina209_get_power_overlimit(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x11, retval);
+    return;
+}
+
+void ina209_set_power_overlimit(uint8_t addr, uint16_t *val) {
+    ina209_Write2ByteReg(addr, 0x11, *val);
+    return;
+}
+
+void ina209_get_bus_voltage_overlimit(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x12, retval);
+    return;
+}
+
+void ina209_set_bus_voltage_overlimit(uint8_t addr, uint16_t *val) {
+    ina209_Write2ByteReg(addr, 0x12, *val);
+    return;
+}
+
+void ina209_get_bus_voltage_underlimit(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x13, retval);
+    return;
+}
+
+void ina209_set_bus_voltage_underlimit(uint8_t addr, uint16_t *val) {
+    ina209_Write2ByteReg(addr, 0x13, *val);
+    return;
+}
+
+void ina209_get_calibration(uint8_t addr, uint16_t *retval) {
+    ina209_Read2ByteReg(addr, 0x16, retval);
+    return;
+}
+
+void ina209_set_calibration(uint8_t addr, uint16_t *val) {
+    ina209_Write2ByteReg(addr, 0x16, *val);
+    return;
+}
+
+void _flip_byte_order(uint16_t *input) {
```

Comment:
This should take an int16 and return a flipped int16. Side effects = bad

Reply:
function has been yeeted